lang/src/org/partiql/lang/ast/ast.kt

```diff
@@ -18,12 +18,33 @@ import com.amazon.ion.*
 import org.partiql.lang.util.*
 import java.util.*
 
+/**
+ * Base type for all AST nodes.  
+ */
+sealed class AstNode : Iterable<AstNode> {
```

Comment:
I'm wondering what's the value of having this implement `Iterable<AstNode>` *and* add a `.children` property?  It strikes me that only one is needed, and I suspect that `.children` slightly better since I may want to use `Iterable<AstNode>` for something else at a later time.

Reply:
the iterator forces a traversal order, depthFirst, while children is more flexible and lets the user define the traversal order. 

Example: `children` is more suitable for finding a subtree of a specific path while the `Iterator` is better for visiting every node applying some operation, or finding all nodes of a certain type